Stats Insights Latest Post Summary: abbreviate high totals

ScoutHarris opened the pull request:
Fixes #10375 

To test:

Suggested hack to show high values: 
In `LatestPostSummaryCell:configure`, right before `self.summaryData = summaryData`, hard code the values, like so:
```
summaryData.viewsValue = 1000
summaryData.likesValue = 1234324
summaryData.commentsValue = 987654
```

- On a site where the latest post has numbers over 999 (or do the hack), go to Stats > Insights.
- Verify the numbers are abbreviated:

![some_big_numbers](https://user-images.githubusercontent.com/1816888/48171287-864f3d80-e2b8-11e8-9b8a-748dce6757a7.png)


NOTE: There are constraint warnings introduced in a previous PR. They'll be fixed with some future black magic.

jklausa commented on WordPress/Classes/ViewRelated/Stats/Double+Stats.swift:
```diff
@@ -0,0 +1,32 @@
+
+extension Double {
+
+    func abbreviatedString() -> String {
```
This is screaming for some comments explaining what this method _actually_ does ;P 

Some explanation of black magic (I have _no idea_ what's going on with that `log10` down there and why it works) would be nice too, but I'll settle for just header docs showing what are example inputs/outputs of this.

ScoutHarris replied:
Haha! Ya know, I totally forgot about comments. You of course are correct. I'll add some!

ScoutHarris commented:
Hey @jklausa . I added a comment block with some examples. Also, I fixed an issue where numbers like 999999 were not being rounded up (it was being displayed as 1000k instead of 1M).

Back to you!
